Objects/listobject.c

```diff
@@ -3637,8 +3637,7 @@ list_ass_subscript_lock_held(PyObject *_self, PyObject *item, PyObject *value)
                 step = -step;
             }
 
-            garbage = (PyObject**)
-                PyMem_Malloc(slicelength*sizeof(PyObject*));
+            garbage = PyMem_New(PyObject *, slicelength);
```

Comment:
It does not overflow. `slicelength` is the length of a slice in existing list. The list already uses at least `slicelength*sizeof(PyObject*)` bytes.